maint_tools/create_issue_from_juint.py

```diff
@@ -0,0 +1,108 @@
+"""Creates or updates an issue if the CI fails. This is useful to keep track of
+scheduled jobs that are failing repeatedly.
+
+This script depends on:
+- `defusedxml` for safer parsing for xml
+- `PyGithub` for interacting with GitHub
+
+The GitHub token only requires the `repo:public_repo` scope are described in
+https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes.
+This scope allows the bot to create and edit its own issues. It is best to use a
+github account that does **not** have commit access to the public repo.
+"""
+
+from pathlib import Path
+import sys
+import argparse
+
+import defusedxml.ElementTree as ET
+from github import Github
+
+parser = argparse.ArgumentParser(
+    description="Create or update issue from JUnit test results from pytest"
+)
+parser.add_argument(
+    "bot_github_token", help="Github token for creating or updating an issue"
+)
+parser.add_argument("ci_name", help="Name of CI run instance")
+parser.add_argument("issue_repo", help="Repo to track issues")
+parser.add_argument("link_to_run", help="URL to link to")
+parser.add_argument("junit_file", help="JUnit file")
+
+args = parser.parse_args()
+gh = Github(args.bot_github_token)
+issue_repo = gh.get_repo(args.issue_repo)
+title = f"⚠️ CI failed on {args.ci_name} ⚠️"
+
+
+def get_issue():
```

Comment:
would it be possible to check the dependency and scikit-learn versions of the open issue against the one in the current CI, and then only return that issue if they match?

Reply:
It is possible, but I think it makes everything a bit more complicated. If we check the dependencies and scikit-learn version, then the `scipy-dev` config may have multiple open issues. For example, if the `scipy-dev config` has updated dependencies, it would open another issue. This happens quite frequently because `scipy-dev` uses the nightly builds. In these cases, we either let the CI close the old issues, or we close it ourselves. 

I prefer the current implementation since it has a one-to-one mapping between a `[scipy-dev]` and an open issue.

- If `[scipy-dev]` is passing -> close issue connected to `[scipy-dev]`
- If `[scipy-dev]` is failing -> update issue connected to `[scipy-dev]` or create a new issue if there isn't an open issue connected to `[scipy-dev]`

Reply:
Fair enough, we can let it be and see how it works. I'd also be happy if we had a ping to `@core-devs` to the first post of each issue. We can then unsubscribe if we want.